src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -98,38 +98,30 @@ public int Take()
 
         public void Skip(int bytesToSkip)
```

Comment:
If I were to change this, I would have it throw if there aren't enough following bytes instead of advancing to the end.

Attempting to skip more bytes than there actually are is indicative of a bug in the caller, and none of our existing code would ever skip too many bytes.